<?xml version="1.0" ?>
<launch>
  <arg name="robot_ip" default="172.16.0.104"/>
  <arg name="load_gripper" default="true" />
  <include file="$(find panda_ros)/launch/franka_control_hazard.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <rosparam command="load" file="$(find panda_ros)/config/panda_ros_controller.yaml" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_impedance_equilibrium_controller"/>
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_example_controllers)/launch/rviz/franka_description_with_marker.rviz"/>
  <node name="equilibrium_publisher" pkg="panda_ros" type="equilibrium_publisher_force_max.py" output="screen" required="true" >
    <param name="link_name" value="panda_link0" />
  </node>

  <node name="message_synchroniser" pkg="panda_ros" type="message_synchroniser.py" output="screen" required="true"/>

  <node name="painting" pkg="panda_ros" type="painting.py" output="screen" required="true"/>
  <node name="goal_handler" pkg="panda_ros" type="goal_handler.py" output="screen" required="true"/>
  <node name="analytics_calc" pkg="panda_ros" type="analytics_calc.py" output="screen" required="true"/>
</launch>
